AniHubApp/AniHubApp/ViewModels/SearchAnimeByGenreViewModel.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using AniHubApp.Models;
+using AniHubApp.Services;
+using Prism.Navigation;
+using Prism.Services;
+
+namespace AniHubApp.ViewModels
+{
+    public class SearchAnimeByGenreViewModel : BaseViewModel, INavigatedAware
+    {
+        public ObservableCollection<Anime> AnimesByGenre { get; set; }
+        public AnimeGenre AnimeSpecificGenre { get; set; }
+
+        private IAniApiService _aniApiService;
+        private IPageDialogService _pageDialogService;
+        public SearchAnimeByGenreViewModel(INavigationService navigationService, IAniApiService aniApiService, IPageDialogService pageDialogService) : base(navigationService)
+        {
+            _pageDialogService = pageDialogService;
+            _aniApiService = aniApiService;
+        }
+        public async void OnNavigatedTo(INavigationParameters parameters)
+        {
+            AnimeSpecificGenre = parameters.GetValue<AnimeGenre>("genre");
+            var queryParams = new AnimeListQueryParams(genre: AnimeSpecificGenre.Name);
```

Comment:
I think the lines of code from 28 to 30 could be a separate function: `GetAnimesByGenre`. Just to keep each function with a single responsiblity